.github/workflows/build_all.yml

```diff
@@ -45,14 +49,19 @@ env:
 
 jobs:
   build_all:
+    env:
+      MACHINE_TYPE: ${{ inputs.machine-type }}
+      BUILD_DIR: >-
+        ${{
+          (inputs.machine-type == 'arm64' && 'full-build-dir-arm64') ||
+                                             'full-build-dir'
+        }}
     runs-on:
       - self-hosted # must come first
       - runner-group=${{ inputs.runner-group }}
       - environment=${{ inputs.runner-env }}
-      - cpu
+      - ${{ env.MACHINE_TYPE }}
```

Comment:
I suspect this should be `${{ inputs.machine-type }}`. Github has inconsistent restrictions on where environment variables can be used.

Reply:
Yeah, I saw a similar discussion here:

https://github.com/actions/runner/issues/2372#issuecomment-1518528105

will fix